src/libraries/src.proj

```diff
@@ -7,6 +7,8 @@
   <ItemGroup>
     <_allSrc Include="$(MSBuildThisFileDirectory)*\src\*.csproj"
              Exclude="@(ProjectExclusions)" />
+    <!-- Include the System.Text.Json.SourceGeneration project -->
+    <_allSrc Include="$(MSBuildThisFileDirectory)System.Text.Json\source_generation\src\System.Text.Json.SourceGeneration.csproj" />
```

Comment:
Should we use the same convention @maryamariyan is establishing here: https://github.com/dotnet/runtime/pull/51064/commits/35a584c530ea9c6cb7668694783ef7fb2d3b5ce6#diff-341288e4401dd8c0e07ec53d05bbffa87a799217327725486fac7ee2922a210eR12

@safern @ViktorHofer might have opinions on the convention used.  I do think we should be consistent throughout the repo.

Reply:
https://github.com/dotnet/runtime/pull/51300#discussion_r614356934.